executor/analyze.go

```diff
@@ -346,30 +350,32 @@ func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultsCh chan<-
 			}
 		}
 		e.wg.Done()
-		if isCloseChanThread {
-			e.wg.Wait()
-			close(resultsCh)
-		}
 	}()
 	for {
-		var ok bool
-		task, ok = <-taskCh
-		if !ok {
-			break
-		}
-		task.job.Start()
-		switch task.taskType {
-		case colTask:
-			resultsCh <- analyzeColumnsPushdown(task.colExec)
-		case idxTask:
-			resultsCh <- analyzeIndexPushdown(task.idxExec)
-		case fastTask:
-			resultsCh <- analyzeFastExec(task.fastExec)
-		case pkIncrementalTask:
-			resultsCh <- analyzePKIncremental(task.colIncrementalExec)
-		case idxIncrementalTask:
-			resultsCh <- analyzeIndexIncremental(task.idxIncrementalExec)
+		select {
+		case task := <-taskCh:
+			task.job.Start()
+			switch task.taskType {
+			case colTask:
+				resultsCh <- analyzeColumnsPushdown(task.colExec)
+			case idxTask:
+				resultsCh <- analyzeIndexPushdown(task.idxExec)
+			case fastTask:
+				resultsCh <- analyzeFastExec(task.fastExec)
+			case pkIncrementalTask:
+				resultsCh <- analyzePKIncremental(task.colIncrementalExec)
+			case idxIncrementalTask:
+				resultsCh <- analyzeIndexIncremental(task.idxIncrementalExec)
+			}
+		default:
+			select {
+			case <-exitCh:
+				return
+			default:
```

Comment:
Can we use one select here?
```
for {
    select {
    case task := <- taskCh:
        ...
    case <-exitCh:
        return
    }
}
```

Reply:
@xuyifangreeneyes  I can use one select If you are not careful about the lost data when exitch is closed.

Reply:
Got it.